packages/wallet/src/pages/transfer/pay.tsx

```diff
@@ -0,0 +1,81 @@
+import { AppLayout } from '@/components/layouts/AppLayout'
+import { Button } from '@/ui/Button'
+import { z } from 'zod'
+import Image from 'next/image'
+import { Form, useZodForm } from '@/ui/forms/Form'
+import { Input } from '@/ui/forms/Input'
+import { Select } from '@/ui/forms/Select'
+
+const paySchema = z.object({
+  fromAccount: z.string(),
+  incomingPaymentPointer: z.string(),
+  amount: z.number(),
```

Comment:
```suggestion
  amount: z.coerce.number(),
```
[Zod coercion](https://zod.dev/?id=coercion-for-primitives)